crates/cairo-profiler/src/trace_reader/function_trace_builder.rs

```diff
@@ -113,16 +119,33 @@ pub fn collect_function_level_profiling_info(
             function_level_config.split_generics,
         );
 
-        let current_call_stack = build_current_call_stack(
+        let mut current_call_stack = build_current_call_stack(
             &call_stack,
             current_function_name,
             function_level_config.show_inlined_functions,
             sierra_statement_idx,
             statements_functions_map.as_ref(),
+            in_syscall_idx,
         );
 
+        // Syscalls have fixed step count, as taken from versioned constants file
+        // That's why we need to add all the steps counted from its trace to the function
+        // that calls the syscall rather than have them added to the syscall itself
+        let calling_function: Vec<FunctionCall> = if in_syscall_idx.is_some() {
```

Comment:
I realised logic is quite convoluted. Essentially what we need to do when we encounter a syscall is:
- add entry to `functions_stack_traces` with 0 steps that corresponds to the syscall call

and that's all since we no longer need to count steps here. We don't even have to remember if we are in a syscall or not, since we will be just adding the same entry to `functions_stack_traces`.

Reply:
Yup, reworked the whole logic to hold syscalls in separate HashMap. I believe it made things simpler :+1: